### PR TITLE
Programming Components, Structure and Keywords #911

### LINKED ISSUE #911: Programming Components, Structure & Keywords
<hr>

## Description 📜

I'll go with the Documentation of the Following Contents below:
1. I'll go over all of the C programming elements.
2. I'll go over every structure and keyword used in C programming.

<hr>

## Domain of Contribution 📊

- [ x] C/CPP

<hr>

### PR DESCRIPTION
**Pull request by Kanishk0412**
<hr>

## Description 📜
I have added Programming Components, Structure and Keywords in C

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #911

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by Kanishk0412** [comment]
@Suvraneel Please review it.

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    ```
+    <br>
+
+* **Comments**
+  <br>
+    Comments are similar to help text in a C programme, but the compiler ignores them. They begin with the characters /* and end with the characters */, as illustrated below.
+
+    ```c
+    /* Program in C */
+    ```
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:33]:
```suggestion
```c
    /* Program in C */
```
```

remove this space for syntax effect from all the code snippets

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    Note - There are no comments within comments, and they don't appear in strings or character literals.
+
+<br>
+
+* **Identifiers**
+<br>
+    A variable, function, or any other user-defined item is identified by a C identifier. An identifier begins with a letter from A to Z, a to z, or an underscore '_', then zero or more letters, underscores, or numbers (0 to 9).
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:40]: don't use `<br>` tags !
Remove all the html tags

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+    Within identifiers, punctuation characters like @, $, and percent are not allowed. The computer language C is case-sensitive. As a result, in C, manpower and manpower are two distinct identifiers. Here are some identifiers that are acceptable:
+
+    ```
+    gwoc      hello     abc    hello_name    a_123
+    gwoc21    _temp     j      a23b9         retVal
+    ```
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:47]: Remove all front space

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    auto	    double  	int	        struct
+    break	    else	    long	    switch
+    case	    enum	    register	typedef
+    char	    extern	    return	    union
+    continue	for	        signed	    void
+    do	        if	        static	    while
+    default	    goto	    sizeof	    volatile
+    const	    float	    short	    unsigned
+    ```
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:63]: Remove all front space

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+<br>
+
+* **Preprocessor Section**
+<br>
+    All header files used in a programme are found in the preprocessing section. It tells the system that the header files should be linked to the system libraries. It is provided by:
+
+    ```c
+    #include<stdio.h>
+    #include<conio.h>
+    ```
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:120]: Remove all front space from all the code snippets

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    return 0;  // return the integer value in the sum  
+}  
+```
+``Output``
+```
+Enter two numbers to be added 8 7
+8 + 7 = 15
+```
+
+
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:248]: REmove all unwanted empty lines

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by Kanishk0412** [comment]
All Changes have been resolved @vasu-1. Please review it once

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
+# Programming Components, Structure & Keywords
+
+## Components and Keywords
+
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:4]: remove this empty line

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+break	    else	    long	    switch
+case	    enum	    register	typedef
+char	    extern	    return	    union
+continue	for	        signed	    void
+do	        if	        static	    while
+default	    goto	    sizeof	    volatile
+const	    float	    short	    unsigned
+```
+
+
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:61]: remove this empty line

File: C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+All header files used in a programme are found in the preprocessing section. It tells the system that the header files should be linked to the system libraries. It is provided by:
+
+```c
+#include<stdio.h>
+#include<conio.h>
+```
+At the time of compilation, the #include declaration incorporates the relevant file as part of a function. As a result, the contents of the included file are compiled with the function. The entirety of the standard input output files, which includes the definitions of stdin, stdout, and stderr, are included in #includestdio.h>. The statement #includestdio.h> must be used whenever the definitions stdin, stdout, and stderr are used in a function.
+
+For diverse uses, numerous header files are provided. # Include math.h>, for example. In a programme, it is used for mathematical functions.
+
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Programming components, Structure & Keyword/Programming Components Structure_Keywords_C.md:117]: remove this empty line

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Remove all unwanted empty lines

**Comment by Kanishk0412** [comment]
@vasu-1 Changes has been resolved.

**Review by vasu-1 — APPROVED** [review]
LGTM @Kanishk0412 

@Suvraneel @Saviour1001

**Pull request merged**